### PR TITLE
MAYA-104839 - Prim specs disappear from Outliner after rename

### PR DESCRIPTION
**Pull request by seando-adsk**
#### MAYA-104839 - Prim specs disappear from Outliner after rename
* Don't send UFE delete when in path change (rename).

### CODE REVIEW

File: lib/mayaUsd/ufe/StagesSubject.cpp [hunk 1]
```diff
@@ -231,7 +231,7 @@ void StagesSubject::stageChanged(UsdNotice::ObjectsChanged const& notice, UsdSta
 			}
 		}
 #if UFE_PREVIEW_VERSION_NUM >= 2015
-		else if (!prim.IsValid())
+		else if (!prim.IsValid() && !InPathChange::inPathChange())
```
Comment (seando-adsk) [inline lib/mayaUsd/ufe/StagesSubject.cpp:234]: Don't send the delete when we are in a path change. The UFE rename notif is sent [here](https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp#L130)

Comment (ppt-adsk) [reply lib/mayaUsd/ufe/StagesSubject.cpp:234]: Shame that we didn't have a failing test to report this.  Feels like we're missing code coverage.

Comment (kxl-adsk) [reply lib/mayaUsd/ufe/StagesSubject.cpp:234]: Indeed, code coverage is high on my wish list. What stops us from adding a test now?

Comment (seando-adsk) [reply lib/mayaUsd/ufe/StagesSubject.cpp:234]: How would we even test this?

Comment (ppt-adsk) [reply lib/mayaUsd/ufe/StagesSubject.cpp:234]: Some kind of rename failure, I would hope.

File: test/lib/ufe/testRename.py [hunk 2]
```diff
@@ -25,6 +25,24 @@
...
 
+class TestObserver(ufe.Observer):
+    def __init__(self):
+        super(TestObserver, self).__init__()
+        self.unexpectedNotif = 0
+        self.renameNotif = 0
+
+    def __call__(self, notification):
+        if isinstance(notification, (ufe.ObjectAdd, ufe.ObjectDelete)):
+            self.unexpectedNotif += 1
```
Comment (seando-adsk) [inline test/lib/ufe/testRename.py:36]: These are the notifs that we should **not** get during a rename.

File: test/lib/ufe/testRename.py [hunk 3]
```diff
@@ -25,6 +25,24 @@
...
+        if isinstance(notification, ufe.ObjectRename):
+            self.renameNotif += 1
```
Comment (seando-adsk) [inline test/lib/ufe/testRename.py:38]: We do expect a rename notif.

File: test/lib/ufe/testRename.py [hunk 4]
```diff
@@ -120,14 +138,14 @@ def testRename(self):
         self.assertTrue(stage.GetRootLayer().GetPrimAtPath("/TreeBase"))
 
         # get default prim
-        defualtPrim = stage.GetDefaultPrim()
-        self.assertEqual(defualtPrim.GetName(), 'TreeBase')
+        defaultPrim = stage.GetDefaultPrim()
```
Comment (seando-adsk) [inline test/lib/ufe/testRename.py:141]: Fix spelling

File: test/lib/ufe/testRename.py [hunk 5]
```diff
@@ -380,3 +398,48 @@ def testRenameSpecialCharacter(self):
...
+        ufe.Scene.addObjectDeleteObserver(ufeObs)
+        ufe.Scene.addObjectPathChangeObserver(ufeObs)
+
+        # rename
+        newName = 'pCylinder1_Renamed'
+        cmds.rename(newName)
+
+        # After the rename we should have 1 rename notif and no unexepected notifs.
+        self.assertEqual(ufeObs.notifications(), 1)
+        self.assertFalse(ufeObs.receivedUnexpectedNotif())
```
Comment (seando-adsk) [inline test/lib/ufe/testRename.py:445]: Here we rename a UFE item. We should get a single UFE rename notif and no unexpected notifs (object add or delete). The test passes now (with my fix) and I ran it without my fix (above) and it failed (as it should).

**Pull request merged**